### PR TITLE
chore: set up mergify and PR labelling

### PR DESCRIPTION
**Pull request by SiTaggart**
Set up Mergify for handling our merge queue better.

Merging will be based on a "ready to merge" label. For some extra protection if it's labelled with "do not merge" it will be ignored regardless of checks and approvals.

Also set up auto labeling to help manage labelling of PRs. New ones get do not merge by default.

### CODE REVIEW

File: .github/mergify.yml
Comment (gloriliale) [inline .github/mergify.yml:26]: What is it here that would make it unable to rebase? Also, is squash here doing a rebase as well?

File: .github/opened-pr-labeler.yml
Comment (gloriliale) [inline .github/opened-pr-labeler.yml:4]: How do we change the labels on PR's to remove "Do Not Merge"?

**Review by gloriliale — APPROVED** [review]
Excited to see how this will revolutionize our lives  🎉

File: .github/opened-pr-labeler.yml
Comment (SiTaggart) [reply .github/opened-pr-labeler.yml:4]: You just remove the label from the right hand side of the PR page.

File: .github/mergify.yml
Comment (SiTaggart) [reply .github/mergify.yml:26]: Not entirely sure honestly. Unless the squash is a merge commit, and not a rebased commit. The docs aren't clear.

**Comment by SiTaggart** [comment]
I'm potentially going to close this, as I'm unsure about Mergify having permission to effectively have merge access. Might be too much of a risk

**Comment by SiTaggart** [comment]
Closing for now, gunna explore auto merge with Github

**Pull request closed without merging**
